glean-core/ios/GleanTests/Metrics/LabeledMetricTests.swift

```diff
@@ -218,4 +218,38 @@ class LabeledMetricTypeTests: XCTestCase {
             XCTAssertEqual(error as! String, "Can not create a labeled version of this metric type")
         }
     }
+
+    func testRapidlyRecreatingLabeledMetricsDoesNotCrash() {
+        // Regression test for bug 1733757.
+        // The underlying map implementation has an upper limit of entries it can handle,
+        // currently set to (1<<15)-1 = 32767.
+        // We used to create a new object every time a label was referenced,
+        // leading to exhausting the available storage in that map, which finally results in a panic.
+
+        let counterMetric = CounterMetricType(
+            category: "telemetry",
+            name: "labeled_nocrash_counter",
+            sendInPings: ["metrics"],
+            lifetime: .application,
+            disabled: false
+        )
+
+        let labeledCounterMetric = try! LabeledMetricType<CounterMetricType>(
+            category: "telemetry",
+            name: "labeled_nocrash",
+            sendInPings: ["metrics"],
+            lifetime: .application,
+            disabled: false,
+            subMetric: counterMetric,
+            labels: ["foo"]
+        )
+
+        // We go higher than the maximum we know of.
+        let maxAttempts = Int32(1 << 16)
```

Comment:
nit: After some extra thought on this, bit shifting isn't the best at readability and understanding the code. I mean, I understand what you are doing here but someone else might come along and wonder "why use shift here?". Did you do this simply because it is explicitly defined as `(1<<15)-1` in Rust or something? I guess I'm just wondering why not do something like `Int32(Int16.max +1)` or something.

Reply:
I can change the comment to call out the maximum.